client/src/utils/schemaValidation.ts

```diff
@@ -0,0 +1,37 @@
+export function replaceIdsWithNull(obj: any, depth: number) {
```

Comment:
[nit] The default value for `depth` could be `-1`, so it wouldn't need to be explicitly provided in every function call
```suggestion
export function replaceIdsWithNull(obj: any, depth = -1) {
```